File: Low Level Design (LLD)/DesignPatterns/1. Creational/Builder/Java/Desktop.java
public class Desktop {
    private String monitor;
    private String mouse;
    private String keyboard;
    private String speaker;
    private String ram;
    private String processor;
    private String motherboard;
    private String graphicsCard;

    public void setMonitor(String monitor) {
        this.monitor = monitor;
    }

    public void setMouse(String mouse) {
        this.mouse = mouse;
    }

    public void setKeyboard(String keyboard) {
        this.keyboard = keyboard;
    }

    public void setSpeaker(String speaker) {
        this.speaker = speaker;
    }

    public void setRam(String ram) {
        this.ram = ram;
    }

    public void setProcessor(String processor) {
        this.processor = processor;
    }

    public void setMotherboard(String motherboard) {
        this.motherboard = motherboard;
    }

    public void setGraphicsCard(String graphicsCard) {
        this.graphicsCard = graphicsCard;
    }

    public void showSpecifications() {
        System.out.println("Desktop Specifications:");
        System.out.println("- Monitor: " + monitor);
        System.out.println("- Mouse: " + mouse);
        System.out.println("- Keyboard: " + keyboard);
        System.out.println("- Speaker: " + speaker);
        System.out.println("- RAM: " + ram);
        System.out.println("- Processor: " + processor);
        System.out.println("- Motherboard: " + motherboard);
        System.out.println("- Graphics Card: " + graphicsCard);
        System.out.println("-----------------------------------");
    }
}

File: Low Level Design (LLD)/DesignPatterns/1. Creational/Builder/Cpp/src/Desktop.cpp
#include "../include/Desktop.hpp"
#include <iostream>

void Desktop::setMonitor(std::string aMonitor) {
    monitor = aMonitor;
}

void Desktop::setMouse(std::string aMouse) {
    mouse = aMouse;
}

void Desktop::setKeyboard(std::string aKeyboard) {
    keyboard = aKeyboard;
}

void Desktop::setSpeaker(std::string aSpeaker) {
    speaker = aSpeaker;
}

void Desktop::setRam(std::string aRam) {
    ram = aRam;
}

void Desktop::setProcessor(std::string aProcessor) {
    processor = aProcessor;
}

void Desktop::setMotherboard(std::string aMotherboard) {
    motherboard = aMotherboard;
}

void Desktop::showSpecifications() {
    std::cout << "----------------------------------------------" << std::endl;
    std::cout << "Monitor: " << monitor << std::endl;
    std::cout << "Mouse: " << mouse << std::endl;
    std::cout << "Keyboard: " << keyboard << std::endl;
    std::cout << "Speaker: " << speaker << std::endl;
    std::cout << "Ram: " << ram << std::endl;
    std::cout << "Processor: " << processor << std::endl;
    std::cout << "Motherboard: " << motherboard << std::endl;
    std::cout << "----------------------------------------------" << std::endl;
}
